modules/xmpp/JingleSessionPC.js

```diff
@@ -2251,18 +2205,18 @@ export default class JingleSessionPC extends JingleSession {
      * Closes the peerconnection.
      */
     close() {
-        this.closed = true;
+        this.state = JingleSessionState.ENDED;
+        this.establishmentDuration = undefined;
 
-        // The signaling layer will remove it's listeners
-        this.signalingLayer.setChatRoom(null);
+        this.modificationQueue.push(finishCallback => {
+            // The signaling layer will remove it's listeners
+            this.signalingLayer.setChatRoom(null);
 
-        // do not try to close if already closed.
-        this.peerconnection
-            && ((this.peerconnection.signalingState
-                    && this.peerconnection.signalingState !== 'closed')
-                || (this.peerconnection.connectionState
-                    && this.peerconnection.connectionState !== 'closed'))
-            && this.peerconnection.close();
+            // do not try to close if already closed.
+            this.peerconnection && this.peerconnection.close();
```

Comment:
we could remove the need for a lot of the checks in the peer connection handlers by just assigning them to null before calling close

Reply:
right